fastqheat/__main__.py

```diff
@@ -20,6 +20,44 @@
 USABLE_CPUS_COUNT = len(os.sched_getaffinity(0))
 
 
+def get_fasterqdump_version() -> Optional[str]:
```

Comment:
Do we actually need 3 different function to do the same thing?

Reply:
| they may call commands with different arguments
@dakq in which case would we need other arguments rather than `--version` to get the version?

|  they process output differently
It is three times the same code except from `.splitlines()[0]`

My suggestion is to merge the functions into one, pass the name of the executable as one an only argument, make return statement like this 
```python
return result.stdout.strip().splitlines()[0] if executable == 'ascp` else result.stdout.strip()
```

Btw why not `result.stdout.splitlines()[1]` which will return ''ascp version 4.4.0.116'?

Reply:
> In which case would we need other arguments rather than --version to get the version?

I'm sure I'd be able to come up with examples of programs that return their versions when called as `program -V/-v` or `program version`.

> It is three times the same code except from `.splitlines()[0]`

_For now_, yes.

> My suggestion is to merge the functions into one, pass the name of the executable as one an only argument, make return statement like this

IMO this kind of branching will make the function harder to extend and maintain, but okay.

> Btw why not result.stdout.splitlines()[1] which will return ''ascp version 4.4.0.116'?

The function replaces `os.popen("aspera --version").read()`, which returns `IBM Aspera CLI version 3.9.0.1326.6985b21`.

Reply:
| I'm sure I'd be able to come up with examples of programs that return their versions when called as program -V/-v or program version.

When we meet a case like this, we will add the second argument to the function to pass -V/-v/version/whatever

| IMO this kind of branching will make the function harder to extend and maintain

I am not sure adding an extra if in the future is so bad that we rather make a duplicate of this function for every possible executable

Reply:
Also
| IMO this kind of branching will make the function harder to extend and maintain,

For now we are not planning to add a lot of new executables if any. And If in the future we find ourselves in a situation where we need more than three branches for different output, we can always rewrite the logic in order to make a base class for the repeatable logic (which is 95% of the function in question) and subclasses with a specific method to manipulate `result.stdout`